src/dmd/backend/dlist.d

```diff
@@ -0,0 +1,571 @@
+/**
+ * Interface to the C linked list type.
+ *
+ * List is a complete package of functions to deal with singly linked
+ * lists of pointers or integers.
+ * Features:
+ *      1. Uses mem package.
+ *      2. Has loop-back tests.
+ *      3. Each item in the list can have multiple predecessors, enabling
+ *         different lists to 'share' a common tail.
+ *
+ * Copyright:   Copyright (C) 1986-1990 by Northwest Software
+ *              Copyright (c) 1999-2016 by Digital Mars, All Rights Reserved
+ * Authors:     $(LINK2 http://www.digitalmars.com, Walter Bright)
+ * License:     $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(DMDSRC backend/tk/_dlist.d)
+ */
+
+module dmd.backend.dlist;
+
+import core.stdc.stdarg;
+import core.stdc.stdio;
+import core.stdc.stdlib;
+import core.stdc.string;
+
+extern (C++):
+
+nothrow @nogc
+{
+
+/* **************** TYPEDEFS AND DEFINES ****************** */
+
+struct LIST
+{
+        /* Do not access items in this struct directly, use the         */
+        /* functions designed for that purpose.                         */
+        LIST* next;             /* next element in list                 */
+        int count;              /* when 0, element may be deleted       */
+        union
+        {       void* ptr;      /* data pointer                         */
+                int data;
+        }
+}
+
+alias LIST* list_t;             /* pointer to a list entry              */
+
+/* FPNULL is a null function pointer designed to be an argument to
+ * list_free().
+ */
+
+alias void function(void*) @nogc nothrow list_free_fp;
+
+enum FPNULL = cast(list_free_fp)null;
+
+/* **************** PUBLIC VARIABLES ********************* */
+
+__gshared
+{
+    int list_inited;         // != 0 if list package is initialized
+    list_t list_freelist;
+    int nlist;
+}
+
+/* **************** PUBLIC FUNCTIONS ********************* */
+
+/********************************
+ * Create link to existing list, that is, share the list with
+ * somebody else.
+ *
+ * Returns:
+ *    pointer to that list entry.
+ */
+
```

Comment:
I'm not sure why you always have a newline between the DDoc and the code.  I recommend removing it, but it's not a problem.

Reply:
The newline is the style commonly used. Others have started using the no newline approach. Besides, I try to minimize diffs on initial conversions.